client/core/exchangeratefetcher.go

```diff
@@ -0,0 +1,297 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package core
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/dex"
+)
+
+const (
+	// DefaultFiatCurrency is the currency for displaying assets fiat value.
+	DefaultFiatCurrency = "USD"
+	// fiatRateRequestInterval is the amount of time between calls to the exchange API.
+	fiatRateRequestInterval = 5 * time.Minute
+	// fiatRateDataExpiry : Any data older than fiatRateDataExpiry will be discarded.
+	fiatRateDataExpiry = 60 * time.Minute
+
+	// Tokens. Used to identify exchange rate source.
+	messari      = "messari"
+	coinpaprika  = "coinpaprika"
+	dcrataDotOrg = "dcrdata"
+)
+
+var (
+	dcrDataURL     = "https://explorer.dcrdata.org/api/exchangerate"
+	coinpaprikaURL = "https://api.coinpaprika.com/v1/tickers/%s"
+	messariURL     = "https://data.messari.io/api/v1/assets/%s/metrics"
```

Comment:
`https://data.messari.io/api/v1/assets/{assetKey}/metrics/market-data` seems more efficient.

Reply:
@buck54321, thanks for looking into this. Just noticed `https://data.messari.io/api/v1/assets/{assetKey}/metrics` returns too many unused information.